### ISSUE TITLE
Guild operation main in subdirectory not working on remote

### ISSUE THREAD

**Issue by samedii** [guildai/guildai#246]
Guild complains that it cannot find the file `operations/prepare.py` but it works if I move the file to the "root".
```
Successfully installed road-segmentation-package-0.0.0                                                                                                                    
Starting unet-r1:prepare on vast as 24fba4d41aff41adb585292fb0c9b5ba                                                                                                     
WARNING: cannot import flags from operations/evaluate: No module named operations/evaluate                                                                      
WARNING: cannot import flags from operations/prepare: No module named operations/prepare                                                                         
WARNING: cannot import flags from operations/train: No module named operations/train                                                                                      
WARNING: cannot import flags from operations/search_lr: No module named operations/search_lr                                                                      
WARNING: cannot import flags from operations/train: No module named operations/train                                                                               
guild: No module named prepare
```

If I add an `__init__.py` file to the `operations` folder then I get another error message
```
WARNING: cannot import flags from /root/.guild/runs/532c78ae6946443f9127f98f4c0a3a39/.guild/job-packages/road_segmentation_package/operations/train.py: Traceback (most recent call last):
  File "/opt/conda/lib/python3.7/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/opt/conda/lib/python3.7/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/opt/conda/lib/python3.7/site-packages/guild/plugins/import_argparse_flags_main.py", line 157, in <module>
    main()
  File "/opt/conda/lib/python3.7/site-packages/guild/plugins/import_argparse_flags_main.py", line 48, in main
    _exec_module(args.mod_path, args.package)
  File "/opt/conda/lib/python3.7/site-packages/guild/plugins/import_argparse_flags_main.py", line 147, in _exec_module
    python_util.exec_script(mod_path, mod_name=_exec_mod_name(package))
  File "/opt/conda/lib/python3.7/site-packages/guild/python_util.py", line 387, in exec_script
    exec(code, script_globals)
  File "/root/.guild/runs/532c78ae6946443f9127f98f4c0a3a39/.guild/job-packages/road_segmentation_package/operations/train.py", line 14, in <module>
    from road_segmentation import train
ModuleNotFoundError: No module named 'road_segmentation'
```

I can't change to a relative import because the root directory does not have an `__init__.py`
```
ImportError: attempted relative import with no known parent package
```

guild.yml
```
- package: road-segmentation-package

- model: unet-r1
  sourcecode:
    - '*.py'
    - 'operations/*.py'
    - guild.yml
    - exclude:
        dir:
          - input
          - tb
          - checkpoints
  operations:
    prepare:
      main: operations/prepare
      flags-import: all
      output-scalars: off
```

**Comment by samedii** [comment]
This works if you add the root path to the python path

**Comment by gar1t** [comment]
I created an resolution doc to illustrate the issues here.

https://github.com/guildai/issue-resolution/tree/master/246-guild-operation-main-in-subdirectory-not-working-on-remote

TLDR; Use `operations.prepare` for your `main` spec. Be certain that `operations/__init__.py` to make the dir a Python package.

**Comment by samedii** [comment]
Thank you!

**Comment by samedii** [comment]
@gar1t Suggestion: Could add a warning or error if a user defined a `main` spec with a slash `/` and suggest using a dot?
```
main: operations.prepare
```

**Comment by gar1t** [comment]
The spec `operations/prepare` is entirely legit. It just means something different. This is how you specify the start of the Python source path for a module. The example shows how you can use this form - it's quite common.

**Comment by gar1t** [comment]
I feel like this could be a linting topic. Guild could scan a project and detect this condition. But that's a pretty expensive test as Guild would need to examine the main module to see how it's importing other modules.

**Comment by samedii** [comment]
Okay I see. I think that detecting the project strutcture sounds like overkill
